### PR TITLE
Editable, Filterable Lists

### PR DESCRIPTION
**Pull request by kenjinp**
# NEW

Changes the project view to show a collection of resource lists. These lists can be added, deleted, and named, and filtered, with persistence maintained  in local storage. 

# TODO

Filtering on lists

### CODE REVIEW

File: src/shared/store/actions/nexus/index.ts [hunk 1]
```diff
@@ -161,7 +302,22 @@ export const fetchResources: ActionCreator<ThunkAction> = (
     try {
       const org: Organization = await nexus.getOrganization(orgLabel);
       const project: Project = await org.getProject(projectLabel);
-      const resources: PaginatedList<Resource> = await project.listResources(
+      const elasticSearchViews: ElasticSearchView[] = await project.listElasticSearchViews();
+      const defaultElasticSearchView: ElasticSearchView = elasticSearchViews[0];
```
Comment (mdupont) [inline src/shared/store/actions/nexus/index.ts]: Not sure the first one in the list will always be the default one. Shouldn't we add getDefaultElasticSearchView() to the sdk and discriminate based on '@id'?

Comment (kenjinp) [reply src/shared/store/actions/nexus/index.ts]: Thanks for adding this feature to the sdK!

File: src/shared/components/Lists/FilterDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        <Input
+          suffix={<Icon type="filter" className="certain-category-icon" />}
+        />
+      </AutoComplete>
+      <AutoComplete
+        className="certain-category-search"
+        dropdownClassName="certain-category-search-dropdown"
+        dropdownMatchSelectWidth={false}
+        onSelect={(value, option) => {
+          // selectSchema(value as string);
```
[the remark this answers is not in the document]
Comment (kenjinp) [reply src/shared/components/Lists/FilterDropdown.tsx]: This is a prop, to be implemented immediately after we take care of this beast

File: src/shared/components/Lists/ListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        orgProjectFilterKey,
+        orgLabel,
+        projectLabel,
+        paginationSettings,
+        query
+      )
+    ),
+});
+
+export default connect(
```
Comment (julienmachon) [inline src/shared/components/Lists/ListItem.tsx]: This should not be happening at the component level, but in the view (or "container" component). Components are stateless and not aware of redux. Looks like this has been merged so needs to be added to the refactor list.

File: src/shared/components/Lists/index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const mapDispatchToProps = (dispatch: any, ownProps: any) => {
+  const orgProjectFilterKey = ownProps.orgLabel + ownProps.projectLabel;
+  return {
+    createList: () => dispatch(createList(orgProjectFilterKey)),
+    initialize: () =>
+      dispatch(initializeProjectList(ownProps.orgLabel, ownProps.projectLabel)),
+  };
+};
+
+export default connect(
```
Comment (julienmachon) [inline src/shared/components/Lists/index.tsx]: Same comment goes for that one

File: src/shared/store/index.ts [hunk 5]
```diff
@@ -32,8 +33,10 @@ try {
 export default function configureStore(
   history: History,
   nexus: Nexus,
-  preloadedState: object = {}
+  preloadedState: any = {}
```
Comment (julienmachon) [inline src/shared/store/index.ts]: This should probably be `RootState`.

**Pull request merged**